File: LeetCode-Python3/3. Longest Substring Without Repeating Characters/3.py
class Solution:
    def lengthOfLongestSubstring(self, s):
        """
        :type s: str
        :rtype: int
        """
        last = {}
        maxLength = 0
        location = 0
        for i, v in enumerate(s):
            # print(last)
            # print(location, maxLength)
            if v not in last or last[v] < location:
                maxLength = max(maxLength, i - location + 1)
            else:
                location = last[v] + 1
            last[v] = i
        return maxLength

File: LeetCode-Python3/3. Longest Substring Without Repeating Characters/3.cpp
class Solution {
public:
    int lengthOfLongestSubstring(string s) {
        vector<int> char_ind(256,-1);
        int len = s.length();
        int max_len = 0;
        int start = -1;
        
        int i;
        for(i = 0;i < len;i++)
        {
            if(char_ind[s[i]] > start)
                start = char_ind[s[i]];
            
            max_len = max(max_len,i-start);
            char_ind[s[i]] = i;
        }
        
        return max_len;
    }
};